Analyzer/analyzer.py

```diff
@@ -99,32 +99,23 @@ def SetCFunc(self,blockcode):#funcname,
         # ROOT.gInterpreter.Declare(self.Cfuncs[funcname])
         ROOT.gInterpreter.Declare(blockcode)
 
-    def makePUWeight(self,year,nvtx):
-        if year == '16':
-            pufile_mc_name="%s/src/PhysicsTools/NanoAODTools/python/postprocessing/data/pileup/pileup_profile_Summer16.root" % os.environ['CMSSW_BASE']
-            pufile_data_name="%s/src/PhysicsTools/NanoAODTools/python/postprocessing/data/pileup/PileupData_GoldenJSON_Full2016.root" % os.environ['CMSSW_BASE']
-        elif year == '17':
-            pufile_data_name="%s/src/PhysicsTools/NanoAODTools/python/postprocessing/data/pileup/PileupHistogram-goldenJSON-13tev-2018-99bins_withVar.root" % os.environ['CMSSW_BASE']
-            pufile_mc_name="%s/src/PhysicsTools/NanoAODTools/python/postprocessing/data/pileup/mcPileup2017.root" % os.environ['CMSSW_BASE']
-        elif year == '18':
-            pufile_data_name="%s/src/PhysicsTools/NanoAODTools/python/postprocessing/data/pileup/PileupHistogram-goldenJSON-13tev-2018-100bins_withVar.root" % os.environ['CMSSW_BASE']
-            pufile_mc_name="%s/src/PhysicsTools/NanoAODTools/python/postprocessing/data/pileup/mcPileup2018.root" % os.environ['CMSSW_BASE']
-
-        puFile_data = ROOT.TFile(pufile_data_name,"READ")
-        puHist_data = puFile_data.Get('pileup')
-        puFile_mc = ROOT.TFile(pufile_mc_name,"READ")
-        puHist_mc = puFile_mc.Get('pu_mc')
-
-        puWeights = puHist_data.Clone()
-        puWeights.Divide(puHist_mc)
-        puWeights.Sumw2()
-
-        ROOT.gInterpreter.ProcessLine("auto puWeights = pileup;")
+    def makePUWeight(self,nvtx):
```

Comment:
This function should probably be standalone (not sure if we need to do it now though). I realized that is relies on SetCFunc but that should also be standalone since it's just calling the gInterpreter and there's no reason it needs to be inside the analyzer class. Perhaps SetCFunc() should be moved entirely to Cscripts.py and makePUweight should be moved to its own script in the Corrections folder.